stdlib/ballerina-builtin/src/main/ballerina/task/natives.bal

```diff
@@ -20,19 +20,20 @@ package ballerina.task;
 @Param {value:"onTrigger: The function which gets called when the timer goes off"}
 @Param {value:"onError: The function that gets called if the onTrigger function returns an error"}
 @Param {value:"schedule: Specifies the initial delay and interval of the timer task"}
+@Param {value:"isDaemon: Specifies whether the timer should run in daemon mode"}
 @Return {value:"The unique ID of the timer task that was scheduled"}
-public native function scheduleTimer ((function() returns error?) onTrigger,
-                                      (function(error) returns ())? onError,
-                                      {int delay = 0; int interval;} schedule) returns string;
+public native function scheduleTimer ((function() returns error?) onTrigger, (function(error) returns ())? onError,
+                                      {int delay = 0; int interval;} schedule, boolean isDaemon) returns string;
```

Comment:
I am not very comfortable with this. For an external user, isDaemon is very confusing.

Reply:
Shall we change the parameter name then?

Reply:
This should be a defaultable param. Also, does the user have to bother about this? Can we always make it a daemon?

Reply:
@lafernando - Can we run these in daemon mode only? WDYT?

Reply:
No .. that is the problem we had originally .. earlier it was effectively running in daemon mode. That is, the timer will get destroyed and the process with it, when the main thread exits. But for timers, we need to generally work even when the main thread exits. This is the case, when users registers timers with a callback, and then they work. But for scenarios as Ballerina Cache and Transactions, it should work in the other way, where as, when the main thread exists, the internal timers used in Cache etc.. should get destroyed, and the process should exit. 

So due to this, the default behavior of timers should be non-daemon, where people can directly use the Timer API, and for other Ballerina features which depend on it, such as Cache etc.. they can pass in the isDaemon property, to make it true, to make it work in the way they want.

Reply:
Let's hold this a bit. I want to discuss and understand further before making this API change.

Reply:
Yes +1 for reviewing .. daemon is a Java concept and we should find an alternative that works for us.